font/src/lib.rs

```diff
@@ -202,6 +217,75 @@ impl Default for RasterizedGlyph {
     }
 }
 
+// Returns a custom underline cursor character
+pub fn get_underline_cursor_glyph(descent: i32, width: i32) -> Result<RasterizedGlyph, Error> {
+    // Create a new rectangle, the height is relative to the font width
+    let height = cmp::max(width * CURSOR_WIDTH_PERCENTAGE / 100, 1);
+    let buf = vec![255u8; (width * height * 3) as usize];
+
+    // Create a custom glyph with the rectangle data attached to it
+    return Ok(RasterizedGlyph {
+        c: UNDERLINE_CURSOR_CHAR,
+        top: descent + height,
+        left: 0,
+        height,
+        width,
+        buf: buf,
+    });
+}
+
+// Returns a custom beam cursor character
+pub fn get_beam_cursor_glyph(
+    ascent: i32,
+    height: i32,
+    width: i32,
+) -> Result<RasterizedGlyph, Error> {
+    // Create a new rectangle that is at least one pixel wide
+    let beam_width = cmp::max(width * CURSOR_WIDTH_PERCENTAGE / 100, 1);
+    let buf = vec![255u8; (beam_width * height * 3) as usize];
+
+    // Create a custom glyph with the rectangle data attached to it
+    return Ok(RasterizedGlyph {
+        c: BEAM_CURSOR_CHAR,
+        top: ascent,
+        left: 0,
+        height,
+        width: beam_width,
+        buf: buf,
+    });
+}
+
+// Returns a custom beam cursor character
```

Comment:
There's a typo in this comment. I have this corrected locally, but I'm waiting on feedback about the box cursor for underline and beam. In case the box cursor change is not desired, this should probably still be corrected.